### PR TITLE
Add driver proxy generator

### PR DESCRIPTION
**Pull request by suanmiao**

### CODE REVIEW

File: databricks/labs/doc_qa/evaluators/templated_evaluator.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 logging.basicConfig(level=logging.INFO)
-logger = logging.getLogger(__name__.split(".")[0])
+logger = logging.getLogger(__name__.split(".")[-1])
```
Comment (nfx) [inline databricks/labs/doc_qa/evaluators/templated_evaluator.py:18]: This hints that you should inject the logger and not look it up 😉

File: databricks/labs/doc_qa/doc_qa_helper.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    model_generator = vLllmOpenAICompletionFormatModelGenerator(
+        url=url,
+        pat_token=pat_token,
+        prompt_formatter=doc_qa_task_prompt_template,
+        batch_size=1,
+        model_name=model_name,
+        format_prompt_func=vicuna_format_prompt_func,
+        concurrency=100,
+    )
+    return model_generator
```
Comment (nfx) [inline databricks/labs/doc_qa/doc_qa_helper.py:60]: nit: inline this

File: databricks/labs/doc_qa/doc_qa_helper.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    model_generator = vLllmOpenAICompletionFormatModelGenerator(
+        url=url,
+        pat_token=pat_token,
+        prompt_formatter=doc_qa_task_prompt_template,
+        batch_size=1,
+        model_name=model_name,
+        format_prompt_func=llama2_format_prompt_func,
+        concurrency=100,
+    )
+    return model_generator
```
Comment (nfx) [inline databricks/labs/doc_qa/doc_qa_helper.py:81]: Inline

File: databricks/labs/doc_qa/doc_qa_helper.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        input_df=input_df, temperature=temperature, max_tokens=max_tokens
+    )
+
+    result_df = generate_result.to_dataframe()
+
+    logging.info(f"Finished generating {len(result_df)} rows, starting evaluation")
+    eval_result = evaluator.run_eval(
+        dataset_df=result_df, concurrency=20, catch_error=True
+    )
+    return eval_result
```
Comment (nfx) [inline databricks/labs/doc_qa/doc_qa_helper.py:97]: Inline?..

File: databricks/labs/doc_qa/model_generators/model_generator.py [hunk 5]
```diff
@@ -525,3 +518,93 @@ def _generate(
...
+            top_p = 1
+        else:
+            top_p = 0.95
+
+        all_formatted_prompts = [
+            self._format_prompt_func(message=message, system_prompt_opt=system_prompt)
+            for message in prompts
+        ]
+
+        import requests
```
Comment (nfx) [inline databricks/labs/doc_qa/model_generators/model_generator.py:566]: Can you add "isort" to CI for this repo? :) it'll sort the imports to be at the top of the file

File: databricks/labs/doc_qa/doc_qa_helper.py (hunk 2 above)
Comment (suanmiao) [reply databricks/labs/doc_qa/doc_qa_helper.py:60]: Updated.

File: databricks/labs/doc_qa/evaluators/templated_evaluator.py (hunk 1 above)
Comment (suanmiao) [reply databricks/labs/doc_qa/evaluators/templated_evaluator.py:18]: Good point! Updated.

File: databricks/labs/doc_qa/doc_qa_helper.py (hunk 3 above)
Comment (suanmiao) [reply databricks/labs/doc_qa/doc_qa_helper.py:81]: Updated.

File: databricks/labs/doc_qa/doc_qa_helper.py (hunk 4 above)
Comment (suanmiao) [reply databricks/labs/doc_qa/doc_qa_helper.py:97]: Updated.

File: databricks/labs/doc_qa/model_generators/model_generator.py (hunk 5 above)
Comment (suanmiao) [reply databricks/labs/doc_qa/model_generators/model_generator.py:566]: Oh, this is intentional, as the file grow I might move the module outside, and don't want to introduce global dependency until people use the module, thus trying to constraint the import only inside the module.

**Pull request merged**